server/monitor/src/main/resources/resources/functions.js

```diff
@@ -0,0 +1,686 @@
+function getServerStats() {
+  $.getJSON('/rest/tservers/serverStats', function(data) {
+    sessionStorage.serverStats = JSON.stringify(data);
+  });
+}
+
+/**
+ * REST GET call for the recovery list, stores it on a sessionStorage variable
+ */
+function getRecoveryList() {
+  $.getJSON('/rest/tservers/recovery', function(data) {
+    sessionStorage.recoveryList = JSON.stringify(data);
+  });
+}
+
+/**
+ * REST GET call for the participating tablet servers,
+ * stores it on a sessionStorage variable
+ *
+ * @param {string} table Table ID
+ */
+function getTableServers(table) {
+  var call = '/rest/tables/' + table;
+  $.getJSON(call, function(data) {
+    sessionStorage.tableServers = JSON.stringify(data);
+  });
+}
+
+/**
+ * REST GET call for the trace summary, stores it on a sessionStorage variable
+ *
+ * @param {string} minutes Number of minutes to display trace summary
+ */
+function getTraceSummary(minutes) {
+  var call = '/rest/trace/summary/' + minutes;
+  $.getJSON(call, function(data) {
+    sessionStorage.traceSummary = JSON.stringify(data);
+  });
+}
+
+/**
+ * REST GET call for the trace type, stores it on a sessionStorage variable
+ *
+ * @param {string} type Type of the trace
+ * @param {string} minutes Number of minutes to display trace
+ */
+function getTraceOfType(type, minutes) {
+  var call = '/rest/trace/listType/' + type + '/' + minutes;
+  $.getJSON(call, function(data) {
+    sessionStorage.traceType = JSON.stringify(data);
+  });
+}
+
+/**
+ * REST GET call for the trace id, stores it on a sessionStorage variable
+ *
+ * @param {string} id Trace ID
+ */
+function getTraceShow(id) {
+  var call = '/rest/trace/show/' + id;
+  $.getJSON(call, function(data) {
+    sessionStorage.traceShow = JSON.stringify(data);
+  });
+}
+
+/**
+ * REST GET call for the logs, stores it on a sessionStorage variable
+ */
+function getLogs() {
+  $.getJSON('/rest/logs', function(data) {
+    sessionStorage.logs = JSON.stringify(data);
+  });
+}
+
+/**
+ * REST POST call to clear logs
+ */
+function clearLogs() {
+  $.post('/rest/logs');
+}
+
+/**
+ * REST GET call for the problems
+ */
+function getProblems() {
+  getProblemSummary();
+  getProblemDetails();
+}
+
+/**
+ * REST POST call to clear all table problems
+ *
+ * @param {string} tableID Table ID
+ */
+function clearTableProblems(tableID) {
+  var call = '/rest/problems/summary?s=' + tableID;
+  // Change plus sign to use ASCII value to send it as a URL query parameter
+  call = call.split('+').join('%2B');
+  $.post(call);
+}
+
+/**
+ * REST POST call to clear detail problems
+ *
+ * @param {string} table Table ID
+ * @param {string} resource Resource for problem
+ * @param {string} type Type of problem
+ */
+function clearDetailsProblems(table, resource, type) {
+  var call = '/rest/problems/details?table=' + table + '&resource=' +
+   resource + '&ptype=' + type;
+  // Changes plus sign to use ASCII value to send it as a URL query parameter
+  call = call.split('+').join('%2B');
```

Comment:
Also, this substitution is to sanitize the table id, not the URL, could sanitize the params individually before concatenation into the URL.